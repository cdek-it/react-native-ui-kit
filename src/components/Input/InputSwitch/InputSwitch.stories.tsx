import { useArgs } from '@storybook/preview-api'
import type { Meta, StoryObj } from '@storybook/react'
import { useCallback } from 'react'

import { InputSwitch } from './InputSwitch'

const meta: Meta<typeof InputSwitch> = {
  title: 'Input/InputSwitch',
  args: { disabled: false, danger: false, checked: false },
  argTypes: { onCheckedChange: { action: 'onCheckedChange' } },
  render: (args) => {
    const [, updateArgs] = useArgs()

    const onCheckedChange = useCallback(
      (nextValue: boolean) => updateArgs({ checked: nextValue }),
      [updateArgs]
    )

    return <InputSwitch {...args} onCheckedChange={onCheckedChange} />
  },
}

export default meta

type Story = StoryObj<typeof InputSwitch>

const InputSwitchStory: Story = {}

export { InputSwitchStory as InputSwitch }
