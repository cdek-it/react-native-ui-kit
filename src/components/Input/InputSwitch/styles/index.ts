export { useHandleStyles } from './useHandleStyles'
export { useSliderStyles } from './useSliderStyles'
