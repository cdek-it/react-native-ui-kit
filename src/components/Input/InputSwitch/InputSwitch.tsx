import { memo, useCallback } from 'react'
import { Pressable, type PressableProps } from 'react-native'
import Animated from 'react-native-reanimated'

import { useHandleStyles, useSliderStyles } from './styles'

export interface InputSwitchProps extends PressableProps {
  checked: boolean
  onCheckedChange?: (checked: boolean) => void
  disabled?: boolean
  danger?: boolean
}

export const InputSwitch = memo<InputSwitchProps>(
  ({
    checked,
    onCheckedChange,
    disabled = false,
    danger = false,
    ...props
  }) => {
    const { sliderStyle, onPressedChange } = useSliderStyles(
      checked,
      disabled,
      danger
    )
    const { handleStyle } = useHandleStyles(checked)

    const handlePress = useCallback(() => {
      onCheckedChange?.(!checked)
    }, [checked, onCheckedChange])

    return (
      <Pressable
        disabled={disabled}
        style={onPressedChange}
        onPress={handlePress}
        {...props}
      >
        <Animated.View style={sliderStyle}>
          <Animated.View style={handleStyle} />
        </Animated.View>
      </Pressable>
    )
  }
)
