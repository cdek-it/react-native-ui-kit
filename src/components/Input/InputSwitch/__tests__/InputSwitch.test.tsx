import { act, render, userEvent } from '@testing-library/react-native'

import { InputSwitch, type InputSwitchProps } from '../InputSwitch'

describe('InputSwitch component tests', () => {
  const inputSnapshotCases = generatePropsCombinations<InputSwitchProps>({
    disabled: [true, false],
    danger: [true, false],
    checked: [true, false],
    testOnly_pressed: [true, false],
  })

  test.each(inputSnapshotCases)(
    'checked - $checked, danger - $danger, disabled - $disabled, pressed - $testOnly_pressed',
    (props) => {
      const renderSwitch = render(<InputSwitch {...props} />)

      expect(renderSwitch.toJSON()).toMatchSnapshot()
    }
  )

  test('Handle press', async () => {
    const mockedOnCheckedChange = jest.fn()

    const { getByTestId } = render(
      <InputSwitch
        checked
        testID='InputSwitch'
        onCheckedChange={mockedOnCheckedChange}
      />
    )
    const pressableContainer = getByTestId('InputSwitch')
    const user = userEvent.setup()

    expect(mockedOnCheckedChange).not.toHaveBeenCalled()

    await act(async () => {
      await user.press(pressableContainer)
    })

    expect(mockedOnCheckedChange).toHaveBeenCalledWith(false)
  })
})
