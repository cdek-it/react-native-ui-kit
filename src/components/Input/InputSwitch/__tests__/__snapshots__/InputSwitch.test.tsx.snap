// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputSwitch component tests checked - false, danger - false, disabled - false, pressed - false 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "height": 21,
      "width": 35,
    }
  }
>
  <View
    collapsable={false}
    jestAnimatedStyle={
      {
        "value": {},
      }
    }
    style={
      {
        "backgroundColor": "#cbcaca",
        "borderColor": "transparent",
        "borderRadius": 100,
        "borderWidth": 1,
        "height": 21,
        "padding": 3.5,
        "width": 35,
      }
    }
  >
    <View
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {},
        }
      }
      style={
        {
          "backgroundColor": "#ffffff",
          "borderRadius": 100,
          "height": 14,
          "left": 2.5,
          "position": "absolute",
          "top": 2.5,
          "width": 14,
        }
      }
    />
  </View>
</View>
`;

exports[`InputSwitch component tests checked - false, danger - false, disabled - false, pressed - true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "height": 21,
      "width": 35,
    }
  }
>
  <View
    collapsable={false}
    jestAnimatedStyle={
      {
        "value": {},
      }
    }
    style={
      {
        "backgroundColor": "rgba(203, 202, 202, 1)",
        "borderColor": "transparent",
        "borderRadius": 100,
        "borderWidth": 1,
        "height": 21,
        "padding": 3.5,
        "width": 35,
      }
    }
  >
    <View
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {},
        }
      }
      style={
        {
          "backgroundColor": "#ffffff",
          "borderRadius": 100,
          "height": 14,
          "left": 2.5,
          "position": "absolute",
          "top": 2.5,
          "width": 14,
        }
      }
    />
  </View>
</View>
`;

exports[`InputSwitch component tests checked - false, danger - false, disabled - true, pressed - false 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "height": 21,
      "width": 35,
    }
  }
>
  <View
    collapsable={false}
    jestAnimatedStyle={
      {
        "value": {},
      }
    }
    style={
      {
        "backgroundColor": "#DFDFDF",
        "borderColor": "transparent",
        "borderRadius": 100,
        "borderWidth": 1,
        "height": 21,
        "padding": 3.5,
        "width": 35,
      }
    }
  >
    <View
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {},
        }
      }
      style={
        {
          "backgroundColor": "#ffffff",
          "borderRadius": 100,
          "height": 14,
          "left": 2.5,
          "position": "absolute",
          "top": 2.5,
          "width": 14,
        }
      }
    />
  </View>
</View>
`;

exports[`InputSwitch component tests checked - false, danger - false, disabled - true, pressed - true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "height": 21,
      "width": 35,
    }
  }
>
  <View
    collapsable={false}
    jestAnimatedStyle={
      {
        "value": {},
      }
    }
    style={
      {
        "backgroundColor": "#DFDFDF",
        "borderColor": "transparent",
        "borderRadius": 100,
        "borderWidth": 1,
        "height": 21,
        "padding": 3.5,
        "width": 35,
      }
    }
  >
    <View
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {},
        }
      }
      style={
        {
          "backgroundColor": "#ffffff",
          "borderRadius": 100,
          "height": 14,
          "left": 2.5,
          "position": "absolute",
          "top": 2.5,
          "width": 14,
        }
      }
    />
  </View>
</View>
`;

exports[`InputSwitch component tests checked - false, danger - true, disabled - false, pressed - false 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "height": 21,
      "width": 35,
    }
  }
>
  <View
    collapsable={false}
    jestAnimatedStyle={
      {
        "value": {},
      }
    }
    style={
      {
        "backgroundColor": "#cbcaca",
        "borderColor": "#e26e63",
        "borderRadius": 100,
        "borderWidth": 1,
        "elevation": 3.5,
        "height": 21,
        "padding": 3.5,
        "shadowColor": "#f6d4d1",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 3.5,
        "width": 35,
      }
    }
  >
    <View
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {},
        }
      }
      style={
        {
          "backgroundColor": "#ffffff",
          "borderRadius": 100,
          "height": 14,
          "left": 2.5,
          "position": "absolute",
          "top": 2.5,
          "width": 14,
        }
      }
    />
  </View>
</View>
`;

exports[`InputSwitch component tests checked - false, danger - true, disabled - false, pressed - true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "height": 21,
      "width": 35,
    }
  }
>
  <View
    collapsable={false}
    jestAnimatedStyle={
      {
        "value": {},
      }
    }
    style={
      {
        "backgroundColor": "rgba(203, 202, 202, 1)",
        "borderColor": "#e26e63",
        "borderRadius": 100,
        "borderWidth": 1,
        "elevation": 3.5,
        "height": 21,
        "padding": 3.5,
        "shadowColor": "#f6d4d1",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 3.5,
        "width": 35,
      }
    }
  >
    <View
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {},
        }
      }
      style={
        {
          "backgroundColor": "#ffffff",
          "borderRadius": 100,
          "height": 14,
          "left": 2.5,
          "position": "absolute",
          "top": 2.5,
          "width": 14,
        }
      }
    />
  </View>
</View>
`;

exports[`InputSwitch component tests checked - false, danger - true, disabled - true, pressed - false 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "height": 21,
      "width": 35,
    }
  }
>
  <View
    collapsable={false}
    jestAnimatedStyle={
      {
        "value": {},
      }
    }
    style={
      {
        "backgroundColor": "#DFDFDF",
        "borderColor": "transparent",
        "borderRadius": 100,
        "borderWidth": 1,
        "height": 21,
        "padding": 3.5,
        "width": 35,
      }
    }
  >
    <View
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {},
        }
      }
      style={
        {
          "backgroundColor": "#ffffff",
          "borderRadius": 100,
          "height": 14,
          "left": 2.5,
          "position": "absolute",
          "top": 2.5,
          "width": 14,
        }
      }
    />
  </View>
</View>
`;

exports[`InputSwitch component tests checked - false, danger - true, disabled - true, pressed - true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "height": 21,
      "width": 35,
    }
  }
>
  <View
    collapsable={false}
    jestAnimatedStyle={
      {
        "value": {},
      }
    }
    style={
      {
        "backgroundColor": "#DFDFDF",
        "borderColor": "transparent",
        "borderRadius": 100,
        "borderWidth": 1,
        "height": 21,
        "padding": 3.5,
        "width": 35,
      }
    }
  >
    <View
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {},
        }
      }
      style={
        {
          "backgroundColor": "#ffffff",
          "borderRadius": 100,
          "height": 14,
          "left": 2.5,
          "position": "absolute",
          "top": 2.5,
          "width": 14,
        }
      }
    />
  </View>
</View>
`;

exports[`InputSwitch component tests checked - true, danger - false, disabled - false, pressed - false 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "height": 21,
      "width": 35,
    }
  }
>
  <View
    collapsable={false}
    jestAnimatedStyle={
      {
        "value": {},
      }
    }
    style={
      {
        "backgroundColor": "#188700",
        "borderColor": "transparent",
        "borderRadius": 100,
        "borderWidth": 1,
        "height": 21,
        "padding": 3.5,
        "width": 35,
      }
    }
  >
    <View
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {},
        }
      }
      style={
        {
          "backgroundColor": "#ffffff",
          "borderRadius": 100,
          "height": 14,
          "left": 16.5,
          "position": "absolute",
          "top": 2.5,
          "width": 14,
        }
      }
    />
  </View>
</View>
`;

exports[`InputSwitch component tests checked - true, danger - false, disabled - false, pressed - true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "height": 21,
      "width": 35,
    }
  }
>
  <View
    collapsable={false}
    jestAnimatedStyle={
      {
        "value": {},
      }
    }
    style={
      {
        "backgroundColor": "rgba(24, 135, 0, 1)",
        "borderColor": "transparent",
        "borderRadius": 100,
        "borderWidth": 1,
        "height": 21,
        "padding": 3.5,
        "width": 35,
      }
    }
  >
    <View
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {},
        }
      }
      style={
        {
          "backgroundColor": "#ffffff",
          "borderRadius": 100,
          "height": 14,
          "left": 16.5,
          "position": "absolute",
          "top": 2.5,
          "width": 14,
        }
      }
    />
  </View>
</View>
`;

exports[`InputSwitch component tests checked - true, danger - false, disabled - true, pressed - false 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "height": 21,
      "width": 35,
    }
  }
>
  <View
    collapsable={false}
    jestAnimatedStyle={
      {
        "value": {},
      }
    }
    style={
      {
        "backgroundColor": "#9A9A9A",
        "borderColor": "transparent",
        "borderRadius": 100,
        "borderWidth": 1,
        "height": 21,
        "padding": 3.5,
        "width": 35,
      }
    }
  >
    <View
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {},
        }
      }
      style={
        {
          "backgroundColor": "#ffffff",
          "borderRadius": 100,
          "height": 14,
          "left": 16.5,
          "position": "absolute",
          "top": 2.5,
          "width": 14,
        }
      }
    />
  </View>
</View>
`;

exports[`InputSwitch component tests checked - true, danger - false, disabled - true, pressed - true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "height": 21,
      "width": 35,
    }
  }
>
  <View
    collapsable={false}
    jestAnimatedStyle={
      {
        "value": {},
      }
    }
    style={
      {
        "backgroundColor": "#9A9A9A",
        "borderColor": "transparent",
        "borderRadius": 100,
        "borderWidth": 1,
        "height": 21,
        "padding": 3.5,
        "width": 35,
      }
    }
  >
    <View
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {},
        }
      }
      style={
        {
          "backgroundColor": "#ffffff",
          "borderRadius": 100,
          "height": 14,
          "left": 16.5,
          "position": "absolute",
          "top": 2.5,
          "width": 14,
        }
      }
    />
  </View>
</View>
`;

exports[`InputSwitch component tests checked - true, danger - true, disabled - false, pressed - false 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "height": 21,
      "width": 35,
    }
  }
>
  <View
    collapsable={false}
    jestAnimatedStyle={
      {
        "value": {},
      }
    }
    style={
      {
        "backgroundColor": "#188700",
        "borderColor": "#e26e63",
        "borderRadius": 100,
        "borderWidth": 1,
        "elevation": 3.5,
        "height": 21,
        "padding": 3.5,
        "shadowColor": "#f6d4d1",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 3.5,
        "width": 35,
      }
    }
  >
    <View
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {},
        }
      }
      style={
        {
          "backgroundColor": "#ffffff",
          "borderRadius": 100,
          "height": 14,
          "left": 16.5,
          "position": "absolute",
          "top": 2.5,
          "width": 14,
        }
      }
    />
  </View>
</View>
`;

exports[`InputSwitch component tests checked - true, danger - true, disabled - false, pressed - true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "height": 21,
      "width": 35,
    }
  }
>
  <View
    collapsable={false}
    jestAnimatedStyle={
      {
        "value": {},
      }
    }
    style={
      {
        "backgroundColor": "rgba(24, 135, 0, 1)",
        "borderColor": "#e26e63",
        "borderRadius": 100,
        "borderWidth": 1,
        "elevation": 3.5,
        "height": 21,
        "padding": 3.5,
        "shadowColor": "#f6d4d1",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 3.5,
        "width": 35,
      }
    }
  >
    <View
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {},
        }
      }
      style={
        {
          "backgroundColor": "#ffffff",
          "borderRadius": 100,
          "height": 14,
          "left": 16.5,
          "position": "absolute",
          "top": 2.5,
          "width": 14,
        }
      }
    />
  </View>
</View>
`;

exports[`InputSwitch component tests checked - true, danger - true, disabled - true, pressed - false 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "height": 21,
      "width": 35,
    }
  }
>
  <View
    collapsable={false}
    jestAnimatedStyle={
      {
        "value": {},
      }
    }
    style={
      {
        "backgroundColor": "#9A9A9A",
        "borderColor": "transparent",
        "borderRadius": 100,
        "borderWidth": 1,
        "height": 21,
        "padding": 3.5,
        "width": 35,
      }
    }
  >
    <View
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {},
        }
      }
      style={
        {
          "backgroundColor": "#ffffff",
          "borderRadius": 100,
          "height": 14,
          "left": 16.5,
          "position": "absolute",
          "top": 2.5,
          "width": 14,
        }
      }
    />
  </View>
</View>
`;

exports[`InputSwitch component tests checked - true, danger - true, disabled - true, pressed - true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "height": 21,
      "width": 35,
    }
  }
>
  <View
    collapsable={false}
    jestAnimatedStyle={
      {
        "value": {},
      }
    }
    style={
      {
        "backgroundColor": "#9A9A9A",
        "borderColor": "transparent",
        "borderRadius": 100,
        "borderWidth": 1,
        "height": 21,
        "padding": 3.5,
        "width": 35,
      }
    }
  >
    <View
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {},
        }
      }
      style={
        {
          "backgroundColor": "#ffffff",
          "borderRadius": 100,
          "height": 14,
          "left": 16.5,
          "position": "absolute",
          "top": 2.5,
          "width": 14,
        }
      }
    />
  </View>
</View>
`;
